addons/sms/wizard/sms_composer.py

```diff
@@ -238,11 +238,14 @@ def _action_send_sms_comment(self, records=None):
     def _action_send_sms_mass(self, records=None):
         records = records if records is not None else self._get_records()
 
-        sms_record_values = self._prepare_mass_sms_values(records)
-        sms_all = self._prepare_mass_sms(records, sms_record_values)
-        if sms_all and self.mass_keep_log and records and isinstance(records, self.pool['mail.thread']):
-            log_values = self._prepare_mass_log_values(records, sms_record_values)
-            records._message_log_batch(**log_values)
+        sms_record_values_filtered = self._filter_sms_values(self._prepare_mass_sms_values(records))
+        records_filtered = records.filtered(lambda record: record.id in sms_record_values_filtered)
+        if self.mass_keep_log and sms_record_values_filtered and isinstance(records_filtered, self.pool['mail.thread']):
+            log_values = self._prepare_mass_log_values(records_filtered, sms_record_values_filtered)
+            mail_messages = records_filtered._message_log_batch(**log_values)
+            for idx, record in enumerate(records_filtered):
+                sms_record_values_filtered[record.id]['mail_message_id'] = mail_messages[idx].id
```

Comment:
Why do we add this value ? don't remember what is the usage of mail_message_id on sms.sms ^^

Reply:
It's ok I guess, on mail.mail it is done automatically thanks to the inherits, I suppose having a clear link for sms is better.